### PR TITLE
CLOUDSTACK-8833: Fixed  Generating url and migrate volume to another storage , resulting two entry in UI and listvolume is not working for that volume

### PR DESCRIPTION
**Pull request by anshul1886**
Update the volume id in volume_store_ref table to newly created volume for migration

### CODE REVIEW

**Comment by bhaisaab** [comment]
LGTM

File: engine/storage/src/org/apache/cloudstack/storage/image/db/VolumeDataStoreDaoImpl.java [hunk 1]
```diff
@@ -350,4 +351,20 @@ public void expireDnldUrlsForZone(Long dcId){
...
+        TransactionLegacy txn = TransactionLegacy.currentTxn();
+        try {
+            VolumeDataStoreVO volumeDataStoreVO = findByVolume(srcVolId);
+            if(volumeDataStoreVO != null) {
+                txn.start();
+                volumeDataStoreVO.setVolumeId(destVolId);
+                update(volumeDataStoreVO.getId(), volumeDataStoreVO);
+                txn.commit();
+            }
+        } catch (Exception e) {
```
Comment (DaanHoogland) [inline engine/storage/src/org/apache/cloudstack/storage/image/db/VolumeDataStoreDaoImpl.java]: please catch more specific/explicit exceptions

Comment (anshul1886) [inline engine/storage/src/org/apache/cloudstack/storage/image/db/VolumeDataStoreDaoImpl.java]: Updated the code to embed the exception in CloudRuntimeException.

**Comment by bhaisaab** [comment]
@anshul1886 ping

**Comment by rhtyd** [comment]
@anshul1886 please rebase against latest master and push -f, update on status of your PR

**Comment by anshul1886** [comment]
@rhtyd Rebased against latest master.

**Comment by priyankparihar** [comment]
Hi Guys, 
I have tested this PR manually (on master branch). Unable to observe mentioned issue. So please consider it for merging.

**Comment by karuturi** [comment]
@priyankparihar BVT results are required. Can you run CI?

**Comment by priyankparihar** [comment]
>Can you run CI?

@karuturi Running.

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 586
 Hypervisor xenserver
 NetworkType Advanced
 Passed=111
 Failed=8
 Skipped=6

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**
* test_network.py

 * test_public_ip_admin_account Failed

* test_deploy_virtio_scsi_vm.py

 * test_01_verify_libvirt Failing since 6 runs

 * test_02_verify_libvirt_after_restart Failing since 6 runs

 * test_03_verify_libvirt_attach_disk Failing since 6 runs

 * test_04_verify_guest_lspci Failing since 6 runs

 * test_05_change_vm_ostype_restart Failing since 6 runs

 * test_06_verify_guest_lspci_again Failing since 6 runs

 * ContextSuite context=TestDeployVirtioSCSIVM>:teardown Failing since 6 runs


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vm_snapshots.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_router_dns.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_routers_network_ops.py
test_disk_offerings.py

**Comment by karuturi** [comment]
@priyankparihar Can you run CI again?
There were failures due to #2066 which is nor merged.

**Comment by priyankparihar** [comment]
@karuturi 
>Can you run CI again?

OK Running.

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 611
 Hypervisor xenserver
 NetworkType Advanced
 Passed=111
 Failed=1
 Skipped=12

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**
* test_router_dns.py

 * test_router_dns_guestipquery Failed


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vm_snapshots.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_routers_network_ops.py
test_disk_offerings.py

**Pull request merged**